### PR TITLE
Add comments to brave_wallet.mojom

### PR DESCRIPTION
**Pull request by bbondy**
Resolves https://github.com/brave/brave-browser/issues/19294

Pls comment on any of these with a replacement string if you have a better initial suggestion. 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/common/brave_wallet.mojom [hunk 1]
```diff
@@ -425,28 +623,68 @@ struct SignMessageRequest {
...
+  // Obtains the user assets for the specified chain ID
   GetUserAssets(string chain_id) => (array<ERCToken> tokens);
+
+  // Adds a user asset for the specified chain ID
   AddUserAsset(ERCToken token, string chain_id) => (bool success);
+
+  // Removes a user asset for the specified chain ID
   RemoveUserAsset(ERCToken token, string chain_id) => (bool success);
+
+  // Sets whether or not an asset is visible
```
Comment (yrliou) [inline components/brave_wallet/common/brave_wallet.mojom:641]: nit: `Sets whether or not an asset is visible for the specified chain ID` to be consistent.

**Review by yrliou — APPROVED** [review]

File: components/brave_wallet/common/brave_wallet.mojom [hunk 2]
```diff
@@ -323,30 +453,59 @@ struct SwitchChainRequest {
...
+// Deals with the ETH JSON RPC API, as well as things like the user's current
+// network.
 interface EthJsonRpcController {
+  // Checks the chain ID for an ethereum chain that should be added
   AddEthereumChain(EthereumChain chain, url.mojom.Url origin) => (string chain_id, bool accepted);
   AddEthereumChainRequestCompleted(string chain_id, bool approved);
-
   GetPendingChainRequests() => (array<EthereumChain> networks);
   GetPendingSwitchChainRequests() => (array<SwitchChainRequest> requests);
+
```
Comment (darkdh) [inline components/brave_wallet/common/brave_wallet.mojom:464]: SwitchChainRequest related API should be grouped together

File: components/brave_wallet/common/brave_wallet.mojom [hunk 3]
```diff
@@ -425,28 +623,68 @@ struct SignMessageRequest {
...
+  // Resets the Ethereum permission for the account
   ResetEthereumPermission(string origin, string account) => (bool success);
+
+  // Obtains the active origin
   GetActiveOrigin() => (string origin);
+
+  // Obtains the pending sign message requests
   GetPendingSignMessageRequests() => (array<SignMessageRequest> requests);
+
```
Comment (darkdh) [inline components/brave_wallet/common/brave_wallet.mojom:687]: SignMessageRequest related API should be grouped together

File: components/brave_wallet/common/brave_wallet.mojom [hunk 4]
```diff
@@ -323,30 +453,59 @@ struct SwitchChainRequest {
...
+// network.
 interface EthJsonRpcController {
+  // Checks the chain ID for an ethereum chain that should be added
   AddEthereumChain(EthereumChain chain, url.mojom.Url origin) => (string chain_id, bool accepted);
   AddEthereumChainRequestCompleted(string chain_id, bool approved);
-
   GetPendingChainRequests() => (array<EthereumChain> networks);
   GetPendingSwitchChainRequests() => (array<SwitchChainRequest> requests);
+
   // UI to native
```
Comment (darkdh) [inline components/brave_wallet/common/brave_wallet.mojom:465]: We can enhance the comment like
"This is used for UI notifying native when the user approve/reject SwitchChainRequest"

File: components/brave_wallet/common/brave_wallet.mojom [hunk 5]
```diff
@@ -425,28 +623,68 @@ struct SignMessageRequest {
...
+  // Resets the Ethereum permission for the account
   ResetEthereumPermission(string origin, string account) => (bool success);
+
+  // Obtains the active origin
   GetActiveOrigin() => (string origin);
+
+  // Obtains the pending sign message requests
   GetPendingSignMessageRequests() => (array<SignMessageRequest> requests);
+
   // UI to native
```
Comment (darkdh) [inline components/brave_wallet/common/brave_wallet.mojom:688]: We can enhance the comment like
"These are used for UI notifying native when the user approve/reject SignMessage[Hardware]Request"

File: components/brave_wallet/common/brave_wallet.mojom (hunk 2 above)
Comment (bbondy) [reply components/brave_wallet/common/brave_wallet.mojom:463]: Updated

File: components/brave_wallet/common/brave_wallet.mojom (hunk 3 above)
Comment (bbondy) [reply components/brave_wallet/common/brave_wallet.mojom]: Updated

File: components/brave_wallet/common/brave_wallet.mojom (hunk 1 above)
Comment (bbondy) [reply components/brave_wallet/common/brave_wallet.mojom]: Updated

File: components/brave_wallet/common/brave_wallet.mojom (hunk 4 above)
Comment (bbondy) [reply components/brave_wallet/common/brave_wallet.mojom]: Updated

File: components/brave_wallet/common/brave_wallet.mojom (hunk 5 above)
Comment (bbondy) [reply components/brave_wallet/common/brave_wallet.mojom]: Updated

**Review by darkdh — APPROVED** [review]

**Pull request merged**